document-generation/igce-document.ts

```diff
@@ -0,0 +1,270 @@
+import exceljs, { Row, Worksheet } from "exceljs";
+import { logger } from "../utils/logging";
+
+import { IndependentGovernmentCostEstimate, IPeriodEstimate, PeriodType } from "../models/document-generation";
+import { INTERNAL_SERVER_ERROR } from "../utils/errors";
+import { ApiBase64SuccessResponse, SuccessStatusCode } from "../utils/response";
+import { convertPeriodToMonths } from "./utils/utils";
+
+export async function generateIGCEDocument(
+  templatePath: string,
+  payload: IndependentGovernmentCostEstimate
+): Promise<ApiBase64SuccessResponse> {
+  if (!payload.periodsEstimate && !payload.fundingDocument && !payload.surgeCapabilities) {
+    return INTERNAL_SERVER_ERROR;
+  }
+
+  // sort the base period and option periods
+  const basePeriodLineItems = payload.periodsEstimate.filter((periodEstimate: IPeriodEstimate) => {
+    return periodEstimate.period.periodType === PeriodType.BASE;
+  });
+  const optionPeriodsLineItems = payload.periodsEstimate.filter((periodEstimate: IPeriodEstimate) => {
+    return periodEstimate.period.periodType === PeriodType.OPTION;
+  });
```

Comment:
We should probably actually sort these by `optionOrder`.

```suggestion
  const basePeriodLineItems = payload.periodsEstimate.filter((periodEstimate: IPeriodEstimate) => {
    return periodEstimate.period.periodType === PeriodType.BASE;
  });
  const optionPeriodsLineItems = payload.periodsEstimate.filter((periodEstimate: IPeriodEstimate) => {
    return periodEstimate.period.periodType === PeriodType.OPTION;
  }).sort((a, b) => a.optionOrder - b.optionOrder);
```